### ISSUE TITLE
holdOn breaking Gulp Watch

### ISSUE THREAD

**Issue by mattfordham** [floatdrop/gulp-batch#10]
Hi there. A recent update to this library has broken Gulp Watch for me. 

For the first run, everything is fine. Subsequent watch triggers don't seem to be fully executing however. 

I traced things down to [Line 30](https://github.com/floatdrop/gulp-batch/blob/master/index.js#L30) and the `holdOn` variable. I also found [the commit](https://github.com/floatdrop/gulp-batch/commit/c75d1ebbbb48566905593f24f54f130944283df7) that broke things.

Basically, things work when `holdOn` is declared as a local variable, but break without the `var` definition. 

I'd be happy to submit a pull request changing it back, but to be honest, I don't quite grasp everything this library is doing and figured it'd be best in someone else's hands.
